lib/mox.ex

```diff
@@ -194,9 +194,14 @@ defmodule Mox do
 
   """
   def defmock(name, options) when is_atom(name) and is_list(options) do
-    behaviour = options[:for] || raise ArgumentError, ":for option is required on defmock"
-    validate_behaviour!(behaviour)
-    define_mock_module(name, behaviour)
+    behaviours = options[:for] || raise ArgumentError, ":for option is required on defmock"
+    mock_funs =
+      behaviours
+      |> Enum.map(&validate_behaviour!/1)
+      |> Enum.reduce(%{}, fn behaviour, acc ->
+        Map.put(acc, behaviour, generate_mock_funs(behaviour))
```

Comment:
My suggestion: make this return functions exclusively, we don't need to know from where they are coming from.